galley/pkg/config/analysis/analyzers/multicluster/meshnetworks.go

```diff
@@ -0,0 +1,82 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package multicluster
+
+import (
+	v1 "k8s.io/api/core/v1"
+
+	"istio.io/api/mesh/v1alpha1"
+	"istio.io/istio/galley/pkg/config/analysis"
+	"istio.io/istio/galley/pkg/config/analysis/analyzers/util"
+	"istio.io/istio/galley/pkg/config/analysis/msg"
+	"istio.io/istio/pkg/config/resource"
+	"istio.io/istio/pkg/config/schema/collection"
+	"istio.io/istio/pkg/config/schema/collections"
+)
+
+// MeshNetworksAnalyzer validates MeshNetworks configuration in multi-cluster.
+type MeshNetworksAnalyzer struct{}
+
+var _ analysis.Analyzer = &MeshNetworksAnalyzer{}
+
+// Metadata implements Analyzer
+func (s *MeshNetworksAnalyzer) Metadata() analysis.Metadata {
+	return analysis.Metadata{
+		Name:        "meshnetworks.MeshNetworksAnalyzer",
+		Description: "Check the validity of MeshNetworks in the cluster",
+		Inputs: collection.Names{
+			collections.IstioMeshV1Alpha1MeshNetworks.Name(),
+			collections.K8SCoreV1Secrets.Name(),
+		},
+	}
+}
+
+// Analyze implements Analyzer
+func (s *MeshNetworksAnalyzer) Analyze(c analysis.Context) {
+	// Service Registies can be Kubernetes(self), MCP and/or remote clusters.
+	serviceRegistries := []string{util.KubernetesServiceRegistry, util.MCPServiceRegistry}
```

Comment:
Can you get it from the source of truth? https://github.com/istio/istio/blob/5a1f13b8682d79fc6ca1ab404defad8da7b66288/pilot/pkg/serviceregistry/providers.go#L21-L30. This is missing some

Reply:
I'm trying to eliminate galley's dependency on pilot code. I am not seeing such dependency, is it okay to depend on pilot?

Reply:
Considering `galley` no longer exists its probably fine. At the very least we need to include all of the providers

Reply:
Done.